eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -557,30 +556,22 @@ class EclairImpl(appKit: Kit) extends Eclair with Logging {
+                                recipient: Either[PublicKey, Sphinx.RouteBlinding.BlindedRoute],
                                 replyPath: Option[Seq[PublicKey]],
                                 userCustomContent: ByteVector)(implicit timeout: Timeout): Future[SendOnionMessageResponse] = {
-    codecs.list(TlvCodecs.genericTlv).decode(userCustomContent.bits) match {
-      case Attempt.Successful(DecodeResult(userCustomTlvs, _)) =>
-        val replyPathId = randomBytes32()
-        val replyRoute = replyPath.map(hops => OnionMessages.buildRoute(randomKey(), hops.map(OnionMessages.IntermediateNode(_)), OnionMessages.Recipient(appKit.nodeParams.nodeId, Some(replyPathId))))
-        OnionMessages.buildMessage(
-          randomKey(),
-          randomKey(),
-          intermediateNodes.map(OnionMessages.IntermediateNode(_)),
-          destination match { case Left(key) => OnionMessages.Recipient(key, None) case Right(route) => OnionMessages.BlindedPath(route) },
-          replyRoute.map(OnionMessagePayloadTlv.ReplyPath(_) :: Nil).getOrElse(Nil),
-          userCustomTlvs) match {
-          case Success((nextNodeId, message)) =>
-            appKit.postman.ask(ref => Postman.SendMessage(nextNodeId, message, replyPath.map(_ => replyPathId), ref, appKit.nodeParams.onionMessageConfig.timeout))(timeout, appKit.system.scheduler.toTyped).mapTo[Postman.OnionMessageResponse].map {
-              case Postman.Response(payload) =>
-                val encodedReplyPath = payload.replyPath_opt.map(route => blindedRouteCodec.encode(route.blindedRoute).require.bytes.toHex)
-                SendOnionMessageResponse(sent = true, None, Some(SendOnionMessageResponsePayload(encodedReplyPath, payload.replyPath_opt.map(_.blindedRoute), payload.records.unknown.map(tlv => tlv.tag.toString -> tlv.value).toMap)))
-              case Postman.NoReply => SendOnionMessageResponse(sent = true, Some("No response"), None)
-              case Postman.SendingStatus(MessageRelay.Sent(_)) => SendOnionMessageResponse(sent = true, None, None)
-              case Postman.SendingStatus(failure: MessageRelay.Failure) => SendOnionMessageResponse(sent = false, Some(failure.toString), None)
-            }
-          case Failure(error) => Future.successful(SendOnionMessageResponse(sent = false, failureMessage = Some(error.toString), response = None))
+    TlvCodecs.tlvStream(MessageOnionCodecs.onionTlvCodec).decode(userCustomContent.bits) match {
+      case Attempt.Successful(DecodeResult(userTlvs, _)) =>
+        val destination = recipient match {
+          case Left(key) => OnionMessages.Recipient(key, None)
+          case Right(route) => OnionMessages.BlindedPath(route)
+        }
+        appKit.postman.ask(ref => Postman.SendMessage(intermediateNodes, destination, replyPath.map(_ :+ appKit.nodeParams.nodeId), userTlvs, ref, appKit.nodeParams.onionMessageConfig.timeout))(timeout, appKit.system.scheduler.toTyped).mapTo[Postman.OnionMessageResponse].map {
```

Comment:
nit: I would expect the user to include themselves in the `replyPath`: if they do, we will end up duplicating the final `nodeId` by adding it ourselves here. Our current [API documentation](https://acinq.github.io/eclair/#messages) doesn't clarify that, we should update it with the next release. I find it slightly more intuitive to ask the user to include their `nodeId` at the end of the `replyPath` when they provide one, but it's your call!

Reply:
Thanks! But you're now duplicating it in this line, it should be:
```suggestion
        appKit.postman.ask(ref => Postman.SendMessage(intermediateNodes, destination, replyPath, userTlvs, ref, appKit.nodeParams.onionMessageConfig.timeout))(timeout, appKit.system.scheduler.toTyped).mapTo[Postman.OnionMessageResponse].map {
```

Reply:
How did I miss that! It should be fixed now.